### DISCUSSION TITLE
Any way to interoperate with cabal/cabal.project only users in haskell-template?

### DISCUSSION THREAD

**Discussion by ParetoOptimalDev** [srid/haskell-template#106] [General]
RIght now if you have a cabal.project with a source-repository-package specified when using haskell-template you'll get an error about git not being found because cabal is trying to clone that source-repository-package as it doesn't have any mechanism to just trust Nix's.

One reason you might find yourself stuck interoperating with cabal.project is because you need to generate darwin binaries, which is not straight-forward to do on Nix at all because (it seems that) [truly static binaries on OSX basically don't exist](https://github.com/NixOS/nixpkgs/issues/227966#issuecomment-1521073421):

Back to cabal though, I believe the root of the issue is because of https://github.com/haskell/cabal/issues/8434#issuecomment-1247035263

One workaround is to modify the cabal.project file and basically remove the source-repository-package before haskell Nix stuff starts like here:

https://github.com/input-output-hk/haskell.nix/issues/1637#issuecomment-1234406878

Also related: https://github.com/input-output-hk/haskell.nix/issues/1637

Edit: looks like I did a more in depth breakdown on this and survey of solutions at: https://github.com/haskell/cabal/issues/8434#issuecomment-1426894404

**Comment by srid** [comment]
I think this can be done. It would involve finishing the `cabal.project` parser at https://github.com/srid/haskell-flake/tree/master/nix/haskell-parsers and then using that to parse and generate source overrides in Nix.

Is this additional feature/complexity worth adding to haskell-flake? That's the question to ask. Probably not, as I like to keep haskell-flake simple and as a thin module on top of what's in nixpkgs. People can just specify these sources as flake inputs and use them.

**Reply by ParetoOptimalDev** [reply]
> Is this additional feature/complexity worth adding to haskell-flake? That's the question to ask. Probably not, as I like to keep haskell-flake simple and as a thin module on top of what's in nixpkgs. People can just specify these sources as flake inputs and use them.

That is what I was doing, until distributing a static OSX binary became necessary on systems where the user doesn't have Nix installed.

Maybe it's not worth it. The best counter-argument is you can't progressively adopt haskell-flake on a codebase because of this, you have to get buy-in for a full switch to nix flakes. 

Whereas if you could adopt it progressively, you could be the guinea pig and work thorugh inevitable pain points you enounter moving over a complex code-base.

That said the cabal parser seems like a good amount of work and interoperating with cabal.project adds a good amount of complexity. I think it depends on whether you believe `cabal.project` should be the source of truth or `flake.nix`. If you believe `cabal.project` should be the source of truth then the added complexity is considered essential complexity.
